### DISCUSSION TITLE
Non riesco a trovare un cliente per nome

### DISCUSSION THREAD

**Discussion by pupiweb** [fattureincloud/api#321] [Q&A] [answered]
Buongiorno
Non riesco a trovare la sintassi giusta per cercare un cliente per nome
La stringa che passo è
"https://api-v2.fattureincloud.it/c/COMPANY_ID/entities/clients?q=name='AAA'"
Non ricevo errore, ma la ricerca non trova il cliente (questo o altri, non dipende dal nome), che invece esiste, e che trovo senza problemi usando FIC
Sicuramente sbaglio qualcosa di stupido, elementare, ma non riesco a vederlo

grazie 

g. pupita

**Comment by valmoz** [comment, marked as answer]
Buongiorno @pupiweb ,
**prima di tutto**: questa è una pagina pubblica, quindi non può condividere dati sensibili, come il company ID e il nome del cliente cercato. Ho provveduto a modificare l'URL per rimuovere queste due informazioni.

**Secondo**: Ho notato che il parametro "q" non è stato sottoposto a URL encoding. Ciò potrebbe portare a una scorretta risoluzione della query, quindi le consigliamo di applicare l'encoding correttamente.

**Terzo**: Il parametro "q" utilizza una grammatica SQL-like, che è molto comune. Se lei utilizza il filtro "=", allora la query avrà successo solo nel caso in cui la stringa sia esattamente identica al campo su cui sta effettuando la ricerca. Ad esempio, se il parametro "q" ha valore "name='Angela'", i clienti restituiti saranno solamente quelli con name esattamente "Angela", NON "angela", "Alberto Angela", "Piero Angela", ecc...
Per ottenere risultati su valori parziali, deve usare gli altri [operatori a disposizione](https://developers.fattureincloud.it/docs/basics/additional-capabilities/filter-results#-building-a-query), come:
- Like (name LIKE '%Angela%')
- Contains  (name CONTAINS 'Angela')
- Starts With  (name STARTS WITH 'Angela')
- Ends With  (name ENDS WITH 'Angela')

Grazie
Mauro

**Reply by pupiweb** [reply]
Ok grazie, gentilissimo ed esauriente :)

g.
